src/backend/drm/surface/gbm.rs

```diff
@@ -102,18 +91,31 @@ where
             Err(err) => return Err((allocator, err.into())),
         };
 
-        if !plane_formats.iter().any(|fmt| fmt.code == code) {
+        let opaque_code = get_opaque(code).unwrap_or(code);
+        if !plane_formats
+            .iter()
+            .any(|fmt| fmt.code == code || fmt.code == opaque_code)
+        {
```

Comment:
Do we always want to prefer a opaque format even if a later format could directly match?

Reply:
Well given users might want to try 10-bit formats first and later 8-bit formats, I would assume so. If the opaque_fallback is allowed, it shouldn't make a difference functionally. That said we need to add a check for underlays before allowing this, that is still missing.

Reply:
Yeah, seems correct. imo we can can just disable underlays completely for now if we select a opaque format for the primary plane during initialization in `DrmCompositor::new`. I want to change the plane selection algorithm anyway.

Reply:
Should I do this in this PR or do you write the updated plane selection code soon anyway?

Reply:
Given that the issue already existed prior to this PR I don't think it has to be solved here. I have no ETA for the plane selection, so I would suggest to create an Issue out of this review comment and I will take care of it. Either by disabling the underlay planes for opaque types or in a new plane assignment algorithm.